config/routes.rb

```diff
@@ -1,3 +1,6 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  namespace :v1, format: :jsonapi do
```

Comment:
This namespacing with v1 is one way to go about it.

An alternative will be to use custom content-types, have you used that before?

Reply:
Yes, you're talking about same route responding to both (for example) html and json, right?

Reply:
no, not really, in the context of jsonapi using links and versioning the api in routes might be tricky.